ReactAndroid/src/main/java/com/facebook/react/modules/storage/AsyncStorageModule.java

```diff
@@ -42,6 +43,7 @@
 
   private ReactDatabaseSupplier mReactDatabaseSupplier;
   private boolean mShuttingDown = false;
+  private Executor executor = Executors.newSingleThreadExecutor();
```

Comment:
I'm not sure if adding a new executor is the best approach here, but If you are adding this you should override the  onCatalystInstanceDestroy method and then shutdown the executor.
See:
https://github.com/facebook/react-native/blob/26684cf3adf4094eb6c405d345a75bf8c7c0bf88/ReactAndroid/src/main/java/com/facebook/react/bridge/NativeModule.java#L51